miner/payload_building_test.go

```diff
@@ -173,6 +174,20 @@ func TestBuildPayload(t *testing.T) {
 	t.Run("with-zero-params", func(t *testing.T) { testBuildPayload(t, true, false, zeroParams) })
 }
 
+func TestDAFilters(t *testing.T) {
+	// Very low max shuld filter all transactions.
+	t.Run("with-tx-filter-max-filters-all", func(t *testing.T) { testDAFilters(t, big.NewInt(1), nil, 0) })
+	t.Run("with-block-filter-max-filters-all", func(t *testing.T) { testDAFilters(t, nil, big.NewInt(1), 0) })
+	// Very high max should filter nothing.
+	t.Run("with-tx-filter-max-too-high", func(t *testing.T) { testDAFilters(t, big.NewInt(1000000), nil, 257) })
+	t.Run("with-block-filter-max-too-high", func(t *testing.T) { testDAFilters(t, nil, big.NewInt(1000000), 257) })
+	// The first transaction has size 100, all other 256 txs are bigger due to random Data, so should get filtered.
+	t.Run("with-tx-filter-all-but-first", func(t *testing.T) { testDAFilters(t, big.NewInt(100), nil, 1) })
+	t.Run("with-block-filter-all-but-first", func(t *testing.T) { testDAFilters(t, nil, big.NewInt(100), 1) })
+	// Zero values for these parameters mean we should mean never filter
```

Comment:
```suggestion
	// Zero values for these parameters means we should never filter
```

Reply:
do we want to repeat this test for `nil, nil`?

Reply:
done